recipe_scrapers/albertheijn.py

```diff
@@ -1,11 +1,64 @@
 # mypy: disallow_untyped_defs=False
 import re
+import inspect
+from typing import Dict, Optional, Tuple, Union
 
 from ._abstract import AbstractScraper
 from ._utils import normalize_string
+from ._schemaorg import SchemaOrg
+
+from recipe_scrapers.settings import settings
+
+import requests
+from bs4 import BeautifulSoup
+
+HEADERS = {
+    'Accept-Language': 'nl',  # ah.nl seems to block any requests not having both these headers.
+    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0'
+}
 
 
 class AlbertHeijn(AbstractScraper):
+    def __init__(
+        self,
+        url: Union[str, None],
+        proxies: Optional[
+            Dict[str, str]
+        ] = None,  # allows us to specify optional proxy server
+        timeout: Optional[
+            Union[float, Tuple[float, float], Tuple[float, None]]
+        ] = None,  # allows us to specify optional timeout for request
+        wild_mode: Optional[bool] = False,
+        html: Union[str, bytes, None] = None,
+    ):
+        if html:
+            self.page_data = html
+            self.url = url
+        else:
+            assert url is not None, "url required for fetching recipe data"
+            resp = requests.get(
+                url,
+                headers=HEADERS,
+                proxies=proxies,
+                timeout=timeout,
+            )
+            self.page_data = resp.content
+            self.url = resp.url
+
+        self.wild_mode = wild_mode
+        self.soup = BeautifulSoup(self.page_data, "html.parser")
+        self.schema = SchemaOrg(self.page_data)
+
+        # attach the plugins as instructed in settings.PLUGINS
+        if not hasattr(self.__class__, "plugins_initialized"):
+            for name, _ in inspect.getmembers(self, inspect.ismethod):
+                current_method = getattr(self.__class__, name)
+                for plugin in reversed(settings.PLUGINS):
+                    if plugin.should_run(self.host(), name):
+                        current_method = plugin.run(current_method)
+                setattr(self.__class__, name, current_method)
+            setattr(self.__class__, "plugins_initialized", True)
```

Comment:
So, on the one hand: thanks for noticing that we'd need to copy this code block to retain the plugin functionality here.  On the other hand: blech, that's annoying to have to copy-paste.

Fortunately I think there is a better way: let's try to follow the approach taken in this scraper:

https://github.com/hhursev/recipe-scrapers/blob/cd0daddb5a52dafe40def2f1a06ee2b5b2957139/recipe_scrapers/bettybossi.py#L34-L43

Translation: if `html` is _not_ provided (so, we need to make a web request), then configure a `requests.Session` with the required settings, use _that_ to retrieve the `html`, and call the existing constructor as-is.

Not a pattern we should use very often, but it does avoid duplication of a lot of code in situations like this.

Reply:
(also another good observation that the `func` variable is unused.  I'll apply that cleanup to the `AbstractScraper` in the main branch)